File: CleverOnion-backend/src/main/java/com/cleveronion/blog/domain/common/DomainEvent.java
package com.cleveronion.blog.domain.common;

/**
 * 领域事件接口
 * 所有领域事件都应该实现此接口
 * 领域事件用于表示领域中发生的重要业务事件
 * 
 * @author CleverOnion
 */
public interface DomainEvent {
    
    /**
     * 获取事件发生的时间戳
     * 
     * @return 事件发生时间（毫秒时间戳）
     */
    long occurredOn();
    
    /**
     * 获取事件类型
     * 
     * @return 事件类型名称
     */
    default String eventType() {
        return this.getClass().getSimpleName();
    }
}

File: CleverOnion-blog-k/src/main/kotlin/com/cleveronion/domain/shared/event/DomainEvent.kt
package com.cleveronion.domain.shared.event

import java.time.Instant

/**
 * 领域事件基接口
 * 
 * 领域事件用于表示在领域中发生的重要业务事件，
 * 这些事件可以被其他聚合或外部系统订阅和处理。
 */
interface DomainEvent {
    /**
     * 事件发生的时间
     */
    val occurredOn: Instant
    
    /**
     * 产生事件的聚合根标识符
     */
    val aggregateId: String
}